internal/eval/eval_test.go

```diff
@@ -698,6 +699,96 @@ func testEvalJSONSET(t *testing.T, store *dstore.Store) {
 	runEvalTests(t, tests, evalJSONSET, store)
 }
 
+func testEvalJSONARRAPPEND(t *testing.T, store *dstore.Store) {
```

Comment:
Add a test case testing the case where the root level object is an array. The implementation which I have provided works for it.
```sh
0.0.0.0:7379> JSON.SET arr $ '[1,2,3,"blue"]'
OK
0.0.0.0:7379> JSON.ARRAPPEND arr $ '["black"]'
1) (integer) 5
0.0.0.0:7379> JSON.GET arr $
"[1,2,3,\"blue\",[\"black\"]]"
0.0.0.0:7379>  JSON.ARRAPPEND arr $ 5
1) (integer) 6
0.0.0.0:7379> JSON.GET arr $
"[1,2,3,\"blue\",[\"black\"],5]"
```